File: Structy_Python/11_linked_list_find.py
'''

    Linkedin List Find

    given a ll, return true if the target exists else return false

    approach: iteration 
        1. loop through the ll 
        2. if value equals to the target
        3. return true
        4. return false at the end 
    
'''


def linked_list_find_iter(head, target):

    while head:
        if head.val == target:
            return True
        head = head.next
    return False


def linked_list_find_recur(head, target):
    if head is None:
        return False

    if head.val == target:
        return True

    return linked_list_find_recur(head.next, target)

File: Structy_JS/11_linkedListFind.js
const linkedListFindIter = ( head, target ) =>
{
    tail = head;

    while ( tail )
    {
        if ( tail.val === target )
        {
            return true;
        }
        tail = tail.next;
    }
    return false;
};


const linkedListFindRecur = ( head, target ) =>
{
    if ( head === null ) return false;
    if ( head.val === target ) return true;

    return linkedListFind( head.next, target );
};